src/Mir/Intrinsics.hs

```diff
@@ -1114,6 +1125,21 @@ readRefPath sym iTypes v = \case
     MirVector_Array <$> readRefPath sym iTypes v path
 
 
+refEq :: IsSymInterface sym =>
+    sym ->
+    MirReference sym tp ->
+    MirReference sym tp ->
+    IO (RegValue sym BoolType)
+refEq sym (MirReference _ _) (MirReference _ _) =
+    -- TODO: implement an equality check for valid references
```

Comment:
Is this a serious limitation for now?

Reply:
I now notice that in the overall summary you mention it's unlikely to cause trouble.